xdsl/irdl.py

```diff
@@ -601,6 +625,10 @@ def get_constraint(
                         (field_name, OptSingleBlockRegionDef()))
                 else:
                     op_def.regions.append((field_name, OptRegionDef()))
+            else:
+                raise PyRDLOpDefinitionError(
+                    f"{field_name} is neither a function, or an "
+                    "operand, result, region, or attribute definition.")
```

Comment:
I would love to have some kind of "how to resolve this error" message here.

Something like "Please make sure to follow the irdl guidelines outlined here (link)".

Can we get something like that? Maybe we can just link to [this notebook](https://github.com/xdslproject/xdsl/blob/main/docs/irdl.ipynb) for now?

Reply:
If we wanted to "go crazy", we could check if an "old" irdl definition was used and print a more specific error message like: "Please use `Annotated[Operand, <type>]` instead of `OperandDef(<type>)`" (if they used an OperandDef, of course).

Reply:
Done (the simple version), tell me what you think!